### PR TITLE
[libjxl][lcms][highway] update to latest version

### PR DESCRIPTION
**Pull request by quyykk**
- #### What does your PR fix?

Updates [libjxl], [lcms] and [highway] to the latest versions. They need to be updated together because libjxl depends on the other two and the old version of libjxl can't be built using the newer version of the others.

Sorry for the duplicated PR, I messed up a bit.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?

same as before

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?

yes

### CODE REVIEW

**Comment by BillyONeal** [comment]
Sorry for cancelling this run, we needed the fleet to pick up a security fix. Restarted...

**Comment by BillyONeal** [comment]
Sorry I'll restart this when the OpenSSL storm has abated a bit...

File: ports/libjxl/portfile.cmake [hunk 1]
```diff
@@ -1,16 +1,13 @@
-set(JPEGXL_VERSION 0.6.1)
+set(JPEGXL_VERSION 0.7.0)
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO libjxl/libjxl
     REF "v${JPEGXL_VERSION}"
```
[the remark this answers is not in the document]
Comment (Cheney-W) [reply ports/libjxl/portfile.cmake:6]: Additional point: Please add `vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}` at the beginning of the file.

**Comment by Cheney-W** [comment]
Are all patch modifications allowed on upstream?

**Comment by Cheney-W** [comment]
Please remove the change related below from baseline.txt:
```
    "shader-slang": {
      "baseline": "0.23.13",
      "port-version": 0
    },
```

File: ports/libjxl/portfile.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (quyykk) [reply ports/libjxl/portfile.cmake:6]: I changed the other ports to use VERSION as well. Hope that was ok.

**Comment by quyykk** [comment]
> Are all patch modifications allowed on upstream?

Which patches are you concerned about? I removed fix-install-directories.patch because it didn't seem to do anything, but all others were upstreamed.

> Please remove the change related below from baseline.txt:

done

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**